app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormComponent.scala

```diff
@@ -65,16 +64,8 @@ case class FormComponent(
 
 object FormComponent {
 
-  //TODO: remove this logic from case class representing data
-  implicit val format: OFormat[FormComponent] = {
-    implicit val formatFieldValue = Json.format[FormComponent]
+  private val templateReads: Reads[FormComponent] = Reads(
+    json => new FormCompomentMaker(json).optFieldValue() fold (us => JsError(us.toString), fv => JsSuccess(fv)))
```

Comment:
Is this the time to correct the typo in the name of FormCompomentMaker, do you think?